### PR TITLE
Update several links

### PR DESCRIPTION
**Pull request by milbrandt**
Updated several links and changed a lot of `http://...` to `https://...`
Further some mdlint warnings in markdowns are fixed.

I tried to give commit Messages which express what they contain.

Remark: In [fcs\docsrc\content\index.md](fcs\docsrc\content\index.md) _Projects using the F# Compiler Services_ are some links which were not reachable, namely
- https://tpetricek.github.io/FSharp.Formatting/
- https://www.fcell.io/
- http://tsunami.io/

### CODE REVIEW

File: DEVGUIDE.md [hunk 1]
```diff
@@ -157,25 +157,25 @@ See the "Debugging The Compiler" section of this [article](https://medium.com/@w
...
- - We use this compiler to compile the source in this distribution, to produce a "proto" compiler, dropped to the `proto` directory. When run, this compiler still relies on `FSharp.Core.dll` with version X.
- - We use the proto compiler to compile the source for `FSharp.Core.dll` in this distribution.
- - We use the proto compiler to compile the source for `FSharp.Compiler.dll`, `fsc.exe`, `fsi.exe`, and other binaries found in this distribution.
+   - We first need an existing F# compiler. We use the one in the `lkg` directory. Let's assume this compiler has an `FSharp.Core.dll` with version X.
+   - We use this compiler to compile the source in this distribution, to produce a "proto" compiler, dropped to the `proto` directory. When run, this compiler still relies on `FSharp.Core.dll` with version X.
+   - We use the proto compiler to compile the source for `FSharp.Core.dll` in this distribution.
+   s- We use the proto compiler to compile the source for `FSharp.Compiler.dll`, `fsc.exe`, `fsi.exe`, and other binaries found in this distribution.
```
Comment (auduchinok) [inline DEVGUIDE.md]: Looks like a typo.

File: fcs/README.md [hunk 2]
```diff
@@ -77,23 +74,21 @@ FAKE is only used to run build.fsx.  Eventually we will likely remove this once
...
-
+Output is in ``docs``.  In the ``FSharp.Compiler.Service`` repo this is checked in and hosted as <https://fsharp.github.io/FSharp.Compiler.Service>.
 
 ## The two other NuGet packages
 
-It also contains both the source, build, packaging and test logic for 
+It also contains both the source, build, packaging and test logic for
 
-* ``FSharp.Compiler.Service.MSBuild.v12`` adds legacy MSBuild v12 support to an instance of FSharp.Compiler.Service, if exact compatibility for scripting references such as ``#r "Foo, Version=1.3.4"`` is required. 
+- ``FSharp.Compiler.Service.MSBuild.v12`` adds legacy MSBuild v12 support to an instance of FSharp.Compiler.Service, if exact compatibility for scripting references such as ``#r "Foo, Version=1.3.4"`` is required.
```
Comment (auduchinok) [inline fcs/README.md]: Why this change is needed?

File: tests/projects/Sample_VS2013_FSharp_ConsoleApp_net40/Program.fs [hunk 3]
```diff
@@ -1,7 +1,7 @@
-﻿// Learn more about F# at http://fsharp.org
+﻿// Learn more about F# at https://fsharp.org
```
Comment (auduchinok) [inline tests/projects/Sample_VS2013_FSharp_ConsoleApp_net40/Program.fs]: Is there any actual benefits in updating the link in test data?

File: DEVGUIDE.md (hunk 1 above)
Comment (milbrandt) [reply DEVGUIDE.md]: yes, thanks

File: fcs/README.md (hunk 2 above)
Comment (milbrandt) [reply fcs/README.md]: it was a complaint of mdlint in Visual Studio Code, where I accepted the proposed change to get rid of it.

https://github.com/DavidAnson/markdownlint/blob/v0.15.0/doc/Rules.md#md004

File: tests/projects/Sample_VS2013_FSharp_ConsoleApp_net40/Program.fs (hunk 3 above)
Comment (milbrandt) [reply tests/projects/Sample_VS2013_FSharp_ConsoleApp_net40/Program.fs]: not in the test data. In several other files are links to fsharp.org (I just found 87 references) so I decided to do a global search&replace for http://fsharp.org which included also the test data. Should do any harm there IMHO.
